### ISSUE TITLE
Ubuntu Software Center (misleadingly) warns that RStudio Desktop "is of bad quality" 

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#1594]
Installing RStudio Desktop v1.1.183 on Ubuntu 16.04 using Ubuntu Software Center results in the alarming message, "The package is of bad quality." However, this is a false alarm, and the user is given the option to proceed with the installation—which completes successfully.

Note: this issue was raised and resolved in v0.99. However, this is also an existing issue with v1.0.153.

<img width="598" alt="ubuntu_rsdektop_bad_quality" src="https://user-images.githubusercontent.com/31009952/31472173-3255a78c-aebb-11e7-8312-bfb1094f711e.png">

**Comment by jmcphers** [comment]
Could you include the Lintian check results? (they probably have some clue as to why the package is getting flagged)

**Comment by ronblum** [comment]
From what I can tell, if you've installed once, subsequent installations will work without the warning. So I need to set up a fresh installation; I'll update when that's ready.

**Comment by JsizzleR** [comment]
Customer reported issue in ticket 23479

**Comment by jmcphers** [comment]
I ran Lintian myself on the deb and got a huge number of errors, reproduced here for posterity:

    E: rstudio: unstripped-binary-or-object usr/lib/rstudio/bin/diagnostics
    E: rstudio: embedded-library usr/lib/rstudio/bin/libQt5Gui.so.5.4.2: libpng
    E: rstudio: embedded-library usr/lib/rstudio/bin/libQt5WebKit.so.5.4.2: sqlite
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc: libyaml
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc: zlib
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc: gmp
    E: rstudio: statically-linked-binary usr/lib/rstudio/bin/pandoc/pandoc
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc-citeproc: libyaml
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc-citeproc: zlib
    E: rstudio: embedded-library usr/lib/rstudio/bin/pandoc/pandoc-citeproc: gmp
    E: rstudio: statically-linked-binary usr/lib/rstudio/bin/pandoc/pandoc-citeproc
    E: rstudio: embedded-library usr/lib/rstudio/bin/plugins/imageformats/libqjp2.so: libjpeg
    E: rstudio: embedded-library usr/lib/rstudio/bin/plugins/imageformats/libqjpeg.so: libjpeg
    E: rstudio: embedded-library usr/lib/rstudio/bin/plugins/imageformats/libqmng.so: libmng
    E: rstudio: embedded-library usr/lib/rstudio/bin/plugins/imageformats/libqtiff.so: tiff
    E: rstudio: unstripped-binary-or-object usr/lib/rstudio/bin/rpostback
    E: rstudio: unstripped-binary-or-object usr/lib/rstudio/bin/rsclang/libclang.so
    E: rstudio: unstripped-binary-or-object usr/lib/rstudio/bin/rsession
    E: rstudio: unstripped-binary-or-object usr/lib/rstudio/bin/rstudio
    E: rstudio: changelog-file-missing-in-native-package
    E: rstudio: no-copyright-file
    E: rstudio: description-starts-with-package-name
    E: rstudio: description-too-short RStudio
    E: rstudio: description-is-pkg-name RStudio
    W: rstudio: extended-description-line-too-long
    W: rstudio: non-standard-dir-perm usr/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/ 0775 != 0755
    W: rstudio: extra-license-file usr/lib/rstudio/COPYING
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/R/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/R/modules/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/R/packages/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/bin/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/bin/pandoc/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/bin/plugins/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/bin/postback/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/bin/rsclang/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/resources/ 0775 != 0755
    W: rstudio: embedded-javascript-library usr/lib/rstudio/resources/grid/datatables/js/jquery.js please use libjs-jquery
    W: rstudio: non-standard-dir-perm usr/lib/rstudio/resources/libclang/ 0775 != 0755
    W: rstudio: extra-license-file usr/lib/rstudio/resources/mathjax-26/LICENSE
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_AMS-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Caligraphic-Bold.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Caligraphic-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Fraktur-Bold.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Fraktur-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Main-Bold.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Main-Italic.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Main-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Math-BoldItalic.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Math-Italic.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Math-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_SansSerif-Bold.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_SansSerif-Italic.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_SansSerif-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Script-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Size1-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Size2-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Size3-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_Typewriter-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_WinChrome-Regular.otf also in fonts-mathjax
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/mathjax-26/fonts/HTML-CSS/TeX/otf/MathJax_WinIE6-Regular.otf also in fonts-mathjax
    E: rstudio: privacy-breach-uses-embedded-file usr/lib/rstudio/resources/mathjax.html You may use libjs-mathjax package. (https://mathjax.rstudio.com/latest/mathjax.js?config=tex-ams-mml_htmlormml)
    E: rstudio: privacy-breach-uses-embedded-file usr/lib/rstudio/resources/presentation/mathjax.html You may use libjs-mathjax package. (https://mathjax.rstudio.com/latest/mathjax.js?config=tex-ams-mml_htmlormml)
    W: rstudio: extra-license-file usr/lib/rstudio/resources/presentation/revealjs/LICENSE
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/presentation/revealjs/fonts/Lato-Bold.ttf also in fonts-lato
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/presentation/revealjs/fonts/Lato-BoldItalic.ttf also in fonts-lato
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/presentation/revealjs/fonts/Lato-Italic.ttf also in fonts-lato
    W: rstudio: duplicate-font-file usr/lib/rstudio/resources/presentation/revealjs/fonts/Lato-Regular.ttf also in fonts-lato
    W: rstudio: package-contains-vcs-control-file usr/lib/rstudio/resources/templates/.gitignore
    W: rstudio: package-contains-vcs-control-file usr/lib/rstudio/www/.gitignore
    W: rstudio: non-standard-dir-perm usr/share/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/applications/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/16x16/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/16x16/apps/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/16x16/mimetypes/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/24x24/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/24x24/apps/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/24x24/mimetypes/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/256x256/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/256x256/apps/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/256x256/mimetypes/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/32x32/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/32x32/apps/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/32x32/mimetypes/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/48x48/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/48x48/apps/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/icons/hicolor/48x48/mimetypes/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/mime/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/mime/packages/ 0775 != 0755
    W: rstudio: non-standard-dir-perm usr/share/pixmaps/ 0775 != 0755
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Positioning.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Svg.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5WebKitWidgets.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Gui.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Qml.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libicudata.so.55
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5WebChannel.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libicuuc.so.55
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Network.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Quick.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Sensors.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Xml.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5XmlPatterns.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5DBus.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Widgets.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Core.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5PrintSupport.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libicui18n.so.55
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5OpenGL.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5Sql.so.5
    E: rstudio: md5sums-lists-nonexistent-file usr/lib/rstudio/bin/libQt5WebKit.so.5
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Positioning.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5PrintSupport.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Qml.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Quick.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Sensors.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Sql.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Svg.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5WebChannel.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5WebKit.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5WebKitWidgets.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Widgets.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5Xml.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libQt5XmlPatterns.so.5.4.2 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libicudata.so.55.1 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libicui18n.so.55.1 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/libicuuc.so.55.1 0755
    E: rstudio: shlib-with-executable-bit usr/lib/rstudio/bin/rsclang/libclang.so 0755

Some of these we can address, but others are outside of our control. For instance, it considers it an error that we include a statically linked copy of Pandoc.

    E: rstudio: statically-linked-binary usr/lib/rstudio/bin/pandoc/pandoc

It also considers it an error that we include our own copy of Mathjax.

    E: rstudio: privacy-breach-uses-embedded-file usr/lib/rstudio/resources/mathjax.html You may use libjs-mathjax package. (https://mathjax.rstudio.com/latest/mathjax.js?config=tex-ams-mml_htmlormml)

I'm presuming that the Software Center will show this message whenever Lintian emits error-level output. Getting to 0 errors is possible but represents quite a lot of installer work.

**Comment by ronblum** [comment]
FWIW, in v0.99, this was addressed in https://github.com/rstudio/rstudio/commit/a80394617be4fb1c2ff15cff920807998466d83f .

Also FWIW, on a fresh installation of Ubuntu 16.04, I wasn't able to reproduce the problem if I only installed it with `Ubuntu Software Center`. I first had to install it with `Ubuntu Software`, a.k.a. `Software Install`. (`Ubuntu Software`/`Software Install `comes standard with Ubuntu; `Ubuntu Software Center `is a separate product.) To reproduce:

1. Make sure you have `Ubuntu Software Center`, which can be installed with
      apt-get install software-center
2. Download the Ubuntu RStudio Desktop 1.1.383 .deb file.
3. Right-click on the file and choose to open it with `Software Install.` Go through the installation process.
4. Right-click on the file and choose to open it with `Ubuntu Software Center`. Go through the installation process. This should yield the "bad quality" message.
